File: app/dbmodels/LinkedContract.php
<?php

$table = array(
    'fields'        => array(),
    'oldfields'     => array(),
    'renamefields'  => array(),
    'indexes'       => array(),
    'oldindexes'    => array(),
    'relationships' => array(),
    'model'         => 'LinkedContract',
    'menu'          => ''
);
$table['fields'] = array(
    'id'          => DBModels::type('increments', 
                                    array('visible' => false)),
    'contract_id' => DBModels::type('integer'),
    'item_type'    => DBModels::type('string', 
                                    array('visible' => false,
                                        'size'    => 100)),
    'item_id'     => DBModels::type('integer', 
                                    array('visible' => false)),
);

File: fusionglpi/database/linked_contract.py
from fusionglpi import db


class LinkedContract(db.Model):
    id = db.Column(db.Integer, primary_key=True)
    contract_id = db.Column(db.Integer, nullable=False, server_default="0")
    item_type = db.Column(db.String(100, None, True))
    item_id = db.Column(db.Integer, nullable=False, server_default="0")
